core/src/epicli/data/common/ansible/playbooks/roles/kubernetes_master/tasks/generate-certificates.yml

```diff
@@ -0,0 +1,205 @@
+      - name: controller-manager.conf
+        kind: ['clientAuth']
+        target: controller-manager.conf
+        parent_ca: ca
+      - name: etcd-healthcheck-client
+        kind: ['clientAuth']
+        target: etcd/healthcheck-client
+        parent_ca: etcd/ca
+      - name: etcd-peer
+        kind: ['serverAuth', 'clientAuth']
+        target: etcd/peer
+        parent_ca: etcd/ca
+      - name: etcd-server
+        kind: ['serverAuth', 'clientAuth']
+        target: etcd/server
+        parent_ca: etcd/ca
+      - name: front-proxy-client
+        kind: ['clientAuth']
+        target: front-proxy-client
+        parent_ca: front-proxy-ca
+      - name: scheduler.conf
+        kind: ['clientAuth']
+        target: scheduler.conf
+        parent_ca: ca
+  block:
+    - name: Set certificates_opt_mapping fact
+      block:
+        - set_fact:
+            certificates_opt_mapping: "{{ certificates_opt_mapping | default([]) + [item] }}"
+          when: certificates_renewal_list is defined and item.name in certificates_renewal_list
+          with_items: "{{ _certificates_opt_mapping }}"
+        - set_fact:
+            certificates_opt_mapping: "{{ _certificates_opt_mapping }}"
+          when: certificates_renewal_list is not defined
+
+    - name: Set backup_dir fact
+      set_fact:
+        backup_dir: >-
+          {{ specification.advanced.certificates.location | regex_replace('\\/$', '') }}-backup-{{ ansible_date_time.iso8601_basic_short }}
+
+    - name: Copy old certificates
+      block:
+        - name: Copy {{ specification.advanced.certificates.location }}
+          synchronize:
+            src: "{{ specification.advanced.certificates.location }}/"
+            dest: "{{ backup_dir }}"
+          delegate_to: "{{ inventory_hostname }}"
+        - name: Copy .conf files with embedded certificates
+          copy:
+            src: "/etc/kubernetes/{{ item }}"
+            dest: "{{ backup_dir }}"
+            remote_src: true
+          with_items:
+            - admin.conf
+            - controller-manager.conf
+            - scheduler.conf
+
+    - name: Ensure necessary directories exist
+      file:
+        path: "{{ item }}"
+        state: directory
+        owner: root
+        group: root
+        mode: u=rw
+      with_items:
+        - "{{ specification.advanced.certificates.location }}/csr"
+        - "{{ specification.advanced.certificates.location }}/ext"
+
+    - name: Generate CSR for kubeadm-managed certificates
+      command: kubeadm alpha certs renew all --csr-only --csr-dir=csr
+      args:
+        chdir: "{{ specification.advanced.certificates.location }}"
+
+    - name: Generate CSR for embedded .conf certificates
+      command: |-
+        openssl req -nodes -newkey rsa:2048 \
+          -keyout csr/{{ item.key }}.key \
+          -out csr/{{ item.key }}.csr \
+          -subj "{{ item.value }}"
+      args:
+        chdir: "{{ specification.advanced.certificates.location }}"
+      with_dict:
```

Comment:
@to-bar could you provide an example with using dict in place like here, not in a separate var?

Reply:
You can keep `with_*` syntax since it's not deprecated yet. More info [here](https://docs.ansible.com/ansible/latest/user_guide/playbooks_loops.html).

For moving from `with_dict ` it seems a var is required, for example:
```
- name: Generate CSR for embedded .conf certificates
  command: |-
    openssl req -nodes -newkey rsa:2048 \
      -keyout csr/{{ item.key }}.key \
      -out csr/{{ item.key }}.csr \
      -subj "{{ item.value }}"
  args:
    chdir: "{{ specification.advanced.certificates.location }}"
  loop: "{{ _certificates|dict2items }}"
  var:
    _certificates:
      # https://kubernetes.io/docs/setup/best-practices/certificates/#configure-certificates-for-user-accounts
      admin.conf: /O=system:masters/CN=kubernetes-admin
      scheduler.conf: /CN=system:kube-scheduler
      controller-manager.conf: /CN=system:kube-controller-manager
```